src/store/movies-store/search.thunk.ts

```diff
@@ -0,0 +1,19 @@
+export const SEARCH = 'SEARCH';
+
+export function search(value: any) {
+  return {type: SEARCH, value};
+}
+
+const initialState = {contents: ['Mirististica', 'Vet'], value: '', works: []};
+
+export default function reducer(state = initialState, action: { type?: any; value?: any; }) {
+  switch(action.type) {
+    case SEARCH: {
+      const {value} = action;
+      const works = state.contents.filter((val) => val.includes(value));
```

Comment:
I would change `val` to something more meaningful, for example to `work`. Now, `val` and `value` a bit confusing.